### PR TITLE
:mortar_board: Topic9 Add an aside for amortization 

### PR DESCRIPTION
**Pull request by jameshughes89**
### Related Issues or PRs
Mentioned in #96 

### What
Add a brief aside presenting the idea of amortized analysis. 

### Why
The topic is outside the scope of the course, but it may be nice to at least have heard of it before actually learning it for realz. 

### Where
Topic9.rst 
Topic9-amortized.rst

### CODE REVIEW

File: src/site/topic9-amortized.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+3. If we try to push again, we will need to call ``expandCapacity``, which is :math:`O(n)`
+
+* Notice how we only needed to call an :math:`O(n)` operation once every :math:`n` constant time :math:`O(1)` calls of ``push``
+* We can think of amortizing/spreading out this :math:`O(n)` work over the :math:`n` constant time calls
+
+.. image:: img/complexity_amortization0.png
+   :width: 500 px
+   :align: center
+
+* Spread out the :math:`O(n)` work over the previous :math:`n` calls to ``push``
```
Comment (twentylemon) [inline src/site/topic9-amortized.rst:53]: I find the formula for total time makes this a little more clear too. Math in text is here, but here I try for _n + 1_ `push` operations:

( n O(1)  +  O(n) ) / ( n + 1 )  = n O(1) / (n+1)  + O(n) / (n+1)   ; each term is still O(1), so overall is O(1)

**Review by twentylemon — APPROVED** [review]

File: src/site/topic9-amortized.rst [hunk 2]
```diff
@@ -46,6 +46,11 @@ Amortization
 * Notice how we only needed to call an :math:`O(n)` operation once every :math:`n` constant time :math:`O(1)` calls of ``push``
 * We can think of amortizing/spreading out this :math:`O(n)` work over the :math:`n` constant time calls
 
+* We think of all these parts as a whole expression
+* :math:`(n O(1) + O(n)) / (n + 1) = n O(1) / (n + 1) + O(n) / (n + 1)`
+* This tells us how much work is being done on average over the :math:`n +1` ``push`` calls
+* If we roughly divide this out, each term comes out to :math:`O(n)`
```
Comment (jameshughes89) [inline src/site/topic9-amortized.rst:52]: Have alooksie to see if this checks out the way you were thinking.

Comment (twentylemon) [reply src/site/topic9-amortized.rst:52]: Looks fine to me. I'd just clarify:

> each term comes out to O(n)

then, all divided by _n_, so the end result is O(1). The way it's written now kinda makes it sound like it's O(n).

**Review by twentylemon — APPROVED** [review]

File: src/site/topic9-amortized.rst (hunk 2 above)
Comment (jameshughes89) [reply src/site/topic9-amortized.rst]: It was a typo. Should have just been `O(1)`

**Pull request merged**